reps/2023-01-09-aws-fleet-support.md

```diff
@@ -0,0 +1,144 @@
+## Summary - AWS Fleet support for ray autoscaler
+
+### General Motivation
+
+Today, AWS autoscaler requires developers to choose their EC2 instance types upfront and codify it in their [autoscaler config](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml#L73-L80). EC2 offers a lot of instances types across different families with varied pricing. Hence, developers can realize cost-savings if their workload does not have a strict dependency on a specific family of instances. [EC2 Fleet](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-fleet.html) is an AWS offering that allows developers to launch a group of instances depending on various parameters like, maximum amount per hour they are willing to pay, supplement their primary on-demand capacity with spot capacity, specify maximum spot prices for each instance, choose from multiple allocation strategies for on-demand and spot capacity etc. This proposal outlines the work needed to support EC2 Fleet on ray autoscaler.
+
+#### Key requirements
+
+-   Autoscaler must offer existing functionalities and hence changes must be backward compatible.
+-   Developers must be able to make use of the features including but not limited to following:
+    -   save costs by combining on-demand capacity with spot capacity
+    -   make use of different spot and on-demand allocation strategies like `price-capacity-optimized`, `diversified`, `lowest-price` etc. to scale up
+    -   specify maximum spot prices for each instance to optimize their costs
+    -   choose optimal instance pools based on availability depending on vCPU and memory requirements instead of having to choose instance type upfront
+-   [Ray autoscaling monitor](https://github.com/ray-project/ray/blob/a03a141c296da065f333ea81445a1b9ad49c3d00/python/ray/autoscaler/_private/monitor.py), [CLI](https://github.com/ray-project/ray/blob/7b4b88b4082297d3790b9e542090228970708270/python/ray/autoscaler/_private/commands.py#L692), [standard autoscaler](https://github.com/ray-project/ray/blob/a03a141c296da065f333ea81445a1b9ad49c3d00/python/ray/autoscaler/_private/autoscaler.py), and placement groups must be able to provision nodes via EC2 fleet.
+-   EC2 Fleet must not interfere with the autoscaler activities.
+-   EC2 Fleet must be supported for both head and worker node types.
+
+### Should this change be within `ray` or outside?
+
+Yes. Specifically within AWS autoscaler.
+
+## Stewardship
+
+### Required Reviewers
+
+-   TBD
```

Comment:
I think myself, @scv119 should be required at least.

Reply:
Let's also add @architkulkarni 

@DmitriGekhtman would love to get your opinion too, but won't block on your review if you're busy

Reply:
added suggested reviewers.